bftclient/test/bft_client_api_tests.cpp

```diff
@@ -362,6 +362,53 @@ TEST_F(ClientApiTestFixture, write_f_plus_one) {
   client.stop();
 }
 
+TEST_F(ClientApiTestFixture, batch_of_writes) {
```

Comment:
This is a great "successful" test case Lior.

However, I don't think it's sufficient to cover this new feature. We need at least one negative test as well that covers failure cases. If there was a behavior for batches similar to the RetryBehavior for single requests, we may have caught the bug with regards to dynamic timeouts in `sendBatch`. Please add a test to ensure retries get sent to all replicas so we don't delay view change. Thanks!

Reply:
Added failure scenario:
1) sendBatch
2)The behaviour does not return replies.
3)Timeout reached
4)catch `BatchTimeoutException`.